src/StreamJsonRpc.Tests/DuplexPipeMarshalingTests.cs

```diff
@@ -786,6 +832,39 @@ public Task AcceptPipeAndChatLater(bool writeOnOdd, IDuplexPipe pipe, Cancellati
         public Task RejectCall(IDuplexPipe pipe) => Task.FromException(new InvalidOperationException("Expected test exception."));
 
         public object ReturnPipeAsObject() => FullDuplexStream.CreatePipePair().Item1;
+
+        public Stream ServerMethodThatReturnsTwoWayStream()
+        {
+            var streamPair = FullDuplexStream.CreatePair();
+            var bytes = Encoding.UTF8.GetBytes("Streamed bits!");
+
+            streamPair.Item1.Write(bytes, 0, bytes.Length);
+            streamPair.Item1.Flush();
+
+            return streamPair.Item2;
+        }
+
+        public StreamContainingClass ServerMethodThatReturnsCustomTypeWithStream()
+        {
+            var streamPair = FullDuplexStream.CreatePair();
+            var stream = new StreamContainingClass(streamPair.Item2);
+
+            var bytes = Encoding.UTF8.GetBytes("More streamed bits!");
+
+            streamPair.Item1.Write(bytes, 0, bytes.Length);
+            streamPair.Item1.Flush();
+
+            return stream;
+        }
+
+        public Stream ServerMethodThatReturnsAndDisposesStream()
+        {
+            var stream = new SimplexStream();
+
+            this.StreamDisposedTask = Task.Run(() => stream.Dispose());
```

Comment:
This introduces a random timing where `stream.Dispose()` may occur anywhere from immediately (before the `return` statement) to arbitrarily later (or when this task is awaited). 
Which ordering do you intend to test? Let's rewrite the test to enforce one particular ordering for a more stable test.

Reply:
This method intended to test returning a stream that the client drops, and it can still be disposed by the server. I'd say that it would be best for the disposal to happen, after we return from this method.
Could I instead save the task under `StreamDisposedTask` and then run it on the client side to guarantee that it runs after the server returns the stream?

Reply:
Yes, that approach would work.

Reply:
I opted for instead testing for stream disposal on RPC disconnect as you suggested in a comment above.